neurom/features/neurite_features.py

```diff
@@ -109,6 +110,19 @@ def number_of_sections_per_neurite(neurites, neurite_type=TreeType.all):
     return (_sec.count(n) for n in neurites if _ttc(neurite_type)(n))
 
 
+@as_neurite_list
+def total_length_per_neurite(neurites, neurite_type=TreeType.all):
+    '''Get an iterable with the total length of a neurite for a given neurite type'''
+    return (sum(_sec.length(ss) for ss in isection(n))
```

Comment:
Although I realise it might be a bit awkward because python isn't very good with anonymous functions, and lambdas are quite limited. So if what I said above is problematic, let me know and I will try to see if it can be simplified.